osu.Game.Tests/Visual/Multiplayer/TestSceneMatchStartControl.cs

```diff
@@ -3,80 +3,155 @@
+        private readonly Bindable<Room> room = new Bindable<Room>();
+
+        private MultiplayerRoom multiplayerRoom;
+        private MultiplayerRoomUser localUser;
+        private OngoingOperationTracker ongoingOperationTracker;
+
+        private PopoverContainer content;
         private MatchStartControl control;
-        private BeatmapSetInfo importedSet;
 
-        private readonly Bindable<PlaylistItem> selectedItem = new Bindable<PlaylistItem>();
+        private OsuButton readyButton => control.ChildrenOfType<OsuButton>().Single();
 
-        private BeatmapManager beatmaps;
-        private RulesetStore rulesets;
+        protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnlyDependencyContainer parent) =>
+            new CachedModelDependencyContainer<Room>(base.CreateChildDependencies(parent)) { Model = { BindTarget = room } };
 
         [BackgroundDependencyLoader]
-        private void load(GameHost host, AudioManager audio)
+        private void load()
         {
-            Dependencies.Cache(rulesets = new RealmRulesetStore(Realm));
-            Dependencies.Cache(beatmaps = new BeatmapManager(LocalStorage, Realm, rulesets, null, audio, Resources, host, Beatmap.Default));
-            Dependencies.Cache(Realm);
+            Dependencies.CacheAs(multiplayerClient.Object);
+            Dependencies.CacheAs(ongoingOperationTracker = new OngoingOperationTracker());
+            Dependencies.CacheAs(availabilityTracker.Object);
+
+            availabilityTracker.SetupGet(a => a.Availability).Returns(beatmapAvailability);
+
+            multiplayerClient.SetupGet(m => m.LocalUser).Returns(() => localUser);
+            multiplayerClient.SetupGet(m => m.Room).Returns(() => multiplayerRoom);
+
+            // By default, the local user is to be the host.
+            multiplayerClient.SetupGet(m => m.IsHost).Returns(() => ReferenceEquals(multiplayerRoom.Host, localUser));
+
+            // Assume all state changes are accepted by the server.
+            multiplayerClient.Setup(m => m.ChangeState(It.IsAny<MultiplayerUserState>()))
+                             .Callback((MultiplayerUserState r) =>
+                             {
+                                 Logger.Log($"Changing local user state from {localUser.State} to {r}");
+                                 localUser.State = r;
+                                 raiseRoomUpdated();
+                             });
+
+            multiplayerClient.Setup(m => m.StartMatch())
+                             .Callback(() =>
+                             {
+                                 multiplayerClient.Raise(m => m.LoadRequested -= null);
+
+                                 // immediately "end" gameplay, as we don't care about that part of the process.
+                                 changeUserState(localUser.UserID, MultiplayerUserState.Idle);
```

Comment:
We do care about that part of the process. It's skipping an entire sample playback plus testing for the disabled state of the button during gameplay. Admittedly, there's no test for it (was probably removed somewhere with the `ClickButtonWhenEnabled` changes, but it needs to happen.

Reply:
Furthermore, this should be stopping the countdown otherwise it gets stuck in this state:
![image](https://user-images.githubusercontent.com/1329837/163298208-99fc5220-c9a8-4de5-b873-e46e0a3387ca.png)

Reply:
Which sample playback do you mean?

From the stuck state, you should be able to cancel or start a new countdown – is that not working?

Reply:
> Which sample playback do you mean?

Hmm, I was referring to the gameplay start sound, but that one's actually being played by the subscreen.

> you should be able to cancel or start a new countdown

Yes, this works.